server/src/storage/uncompressed_data_manager.rs

```diff
@@ -0,0 +1,345 @@
+/* Copyright 2022 The ModelarDB Contributors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+//! Support for managing all uncompressed data that is ingested into the [`StorageEngine`].
+
+use std::collections::{HashMap, VecDeque};
+use std::path::PathBuf;
+
+use paho_mqtt::Message;
+use tracing::{info, info_span};
+
+use crate::storage::data_point::DataPoint;
+use crate::storage::segment::{FinishedSegment, SegmentBuilder};
+
+const UNCOMPRESSED_RESERVED_MEMORY_IN_BYTES: usize = 5000;
+
+/// Converts raw MQTT messages to uncompressed data points and stores uncompressed data points
+/// temporarily in an in-memory buffer that spills to Apache Parquet files. When finished the data
+/// is made available for compression.
+pub struct UncompressedDataManager {
+    /// Path to the folder containing all uncompressed data managed by the [`StorageEngine`].
+    data_folder_path: PathBuf,
+    /// The uncompressed segments while they are being built.
+    uncompressed_data: HashMap<String, SegmentBuilder>,
+    /// Prioritized queue of finished segments that are ready for compression.
```

Comment:
Maybe add how the `FinishedSegment`s are prioritized.

Reply:
Replaced `Prioritized` with `FIFO`.